adapters/adhese/adhese.go

```
@@ -0,0 +1,247 @@
+		return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed as generic Adhese bid.", string(response.Body)))}
+	}
+
+	var adheseBid = adheseBidResponseArray[0]
+
+	if adheseBid.Origin == "JERLICIA" {
+		var extArray []AdheseExt
+		var originDataArray []AdheseOriginData
+		if err := json.Unmarshal(response.Body, &extArray); err != nil {
+			return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed to JERLICIA ext.", string(response.Body)))}
+		}
+
+		if err := json.Unmarshal(response.Body, &originDataArray); err != nil {
+			return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed to JERLICIA origin data.", string(response.Body)))}
+		}
+		bidResponse = convertAdheseBid(adheseBid, extArray[0], originDataArray[0])
+	} else {
+		bidResponse = convertAdheseOpenRtbBid(adheseBid)
+	}
+
+	price, _ := strconv.ParseFloat(adheseBid.Extension.Prebid.Cpm.Amount, 64)
+	width, _ := strconv.ParseUint(adheseBid.Width, 10, 64)
+	height, _ := strconv.ParseUint(adheseBid.Height, 10, 64)
+	bidResponse.Cur = adheseBid.Extension.Prebid.Cpm.Currency
+	if len(bidResponse.SeatBid) > 0 && len(bidResponse.SeatBid[0].Bid) > 0 {
+		bidResponse.SeatBid[0].Bid[0].Price = price
+		bidResponse.SeatBid[0].Bid[0].W = width
+		bidResponse.SeatBid[0].Bid[0].H = height
+	}
+
+	bidderResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	if len(bidResponse.SeatBid) == 0 {
+		return nil, []error{WrapError("Response resulted in an empty seatBid array.")}
+	}
+
+	var errs []error
+	for _, sb := range bidResponse.SeatBid {
+		for i := 0; i < len(sb.Bid); i++ {
+			bid := sb.Bid[i]
+			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: getBidType(bid.AdM),
+			})
+
+		}
+	}
+	return bidderResponse, errs
+}
+
+func convertAdheseBid(adheseBid AdheseBid, adheseExt AdheseExt, adheseOriginData AdheseOriginData) openrtb.BidResponse {
+	adheseExtJson, _ := json.Marshal(adheseOriginData)
```

Please do not ignore the error from `json.Marshal`.